test/Constraints/result_builder_diags.swift

```diff
@@ -657,7 +657,10 @@ struct MyView {
     } // expected-error {{expected identifier after '.' expression}}
   }
 
-  @TupleBuilder var invalidCaseWithoutDot: some P {
+  // FIXME [OPAQUE SUPPORT]: We definitely need a better error that doesn't
+  // include 'τ_0_0'. This error comes from generating a
+  // `DefaultGenericArgument` fix in `TypeVariableBinding::fixForHole`.
```

Comment:
We should be able to fix this by moving the "could not infer underlying type for opaque type" diagnostic into the constraint system and recording that fix instead of the missing generic arguments fix when we default the type variable to a hole. We don't need to do that in this PR, but let's at least file a bug for that task so we don't lose track. Might even be a good starter bug!

Reply:
Perturbing the locators a bit when moving the opening logic into `matchResultBuilder` seems to have stopped this fix from being generated. Now the error messages look the same as before this branch. I could look in to why that is further, but I'm not sure if it's worth the time investment.